### PR TITLE
DTSAM-236 :Challenged Access for Fee-Paid Judicial members

### PR DESCRIPTION
**Pull request by prashanthkumarnb**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSAM-236

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[* ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/ChallengedAccessDroolsTest.java [hunk 1]
```diff
@@ -549,4 +549,123 @@ void shouldRejectAccessFor_ChallengedAccess_ExistingPastEndTime(String roleName,
...
+
+        FeatureFlag featureFlag = FeatureFlag.builder().flagName(FeatureFlagEnum.SSCS_CHALLENGED_1_0.getValue())
+            .status(true).build();
+
+        featureFlags.add(featureFlag);
+
+        HashMap<String, JsonNode> existingAttributes = new HashMap<>();
+        existingAttributes.put("jurisdiction", convertValueJsonNode(jurisdiction));
+        existingAttributes.put("caseType", convertValueJsonNode(caseDetails.getCaseTypeId()));
+        existingAttributes.put("substantive", convertValueJsonNode("Y"));
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/ChallengedAccessDroolsTest.java:590]: Can you set `substantive` to `"N"` to reflect conditions in PROD. thanks

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/ChallengedAccessDroolsTest.java [hunk 2]
```diff
@@ -549,4 +549,123 @@ void shouldRejectAccessFor_ChallengedAccess_ExistingPastEndTime(String roleName,
...
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/ChallengedAccessDroolsTest.java:656]: Can you set `substantive` to `"N"` to reflect conditions in PROD. thanks

**Review by mattnayler — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/ChallengedAccessDroolsTest.java (hunk 1 above)
Comment (prashanthkumarnb) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/ChallengedAccessDroolsTest.java]: done

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/ChallengedAccessDroolsTest.java (hunk 2 above)
Comment (prashanthkumarnb) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/ChallengedAccessDroolsTest.java]: done

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/ChallengedAccessDroolsTest.java (hunk 1 above)
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/ChallengedAccessDroolsTest.java]: Resolved in 68ce2df4398504e24354f7bdd011fd09797c323c.

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/ChallengedAccessDroolsTest.java (hunk 2 above)
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/ChallengedAccessDroolsTest.java]: Resolved in 68ce2df4398504e24354f7bdd011fd09797c323c.

**Review by anthonydummer — APPROVED** [review]

**Review by mattnayler — APPROVED** [review]
LGTM

**Review by mattnayler — APPROVED** [review]
Re-reviewed after bf9068abbe204b588c80e98792213e39a681db3b.

**Review by mikebrownccd — APPROVED** [review]
LGTM

**Review by anthonydummer — APPROVED** [review]
LGTM

**Pull request merged**